docs/guides/ingesters-with-wal.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "Ingesters with WAL"
+linkTitle: "Ingesters with WAL"
+weight: 5
+slug: ingesters-with-wal
+---
+
+Currently the ingesters without the TSDB stores all the data in the memory and if it happens to crash for any reason, there will be a loss of data. WAL helps fill this gap of reliability.
+
+To use WAL, there are some changes that needs to be made in the deployment.
+
+## Things to change
+
+1. Since ingesters need to have the same persistent volume across restarts/rollout, all the ingesters should be run on [statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) with fixed volumes.
+
+2. Following flags needs to be set
+    * `--ingester.wal-dir` to the directory where the WAL data should be stores and/or recovered from.
+    * `--ingester.wal-enabled` to `true` which enables writing to WAL during ingestion.
+    * `--ingester.checkpoint-enabled` to `true` to enable checkpointing of in-memory chunks to disk. This is optional which helps in speeding up the replay process.
+    * `--ingester.checkpoint-duration` to the interval at which checkpoints should be created.
+    * `--ingester.recover-from-wal` to `true` to recover data from an existing WAL. The data is recovered even if WAL is disabled and this is set to `true`. The WAL dir needs to be set for this.
+        * If you are going to enable WAL, it is advisable to always set this to `true`.
+
+## Stuff that is changed automatically when WAL is enabled
+
+1. Flushing of data to chunk store during rollouts or scale down is disabled. This is because during a rollout of statefulset there is no 1 ingester leaving and joining each at the same time, rather the same ingester is shut down and broght back again with updated config. Hence flushing is skipped and the data is recovered from the WAL.
+
+2. As there are no transfers between ingesters, the tokens are stored and recovered from disk between rollout/restarts. This is [not a new thing](https://github.com/cortexproject/cortex/pull/1750) but it is effective when using statefulsets.
```

Comment:
Don't we need to set the tokens path too? And make sure it's on the persisted disk as well? Can you add it to the list of flags to set?

Reply:
Will add.